starcheck/src/lib/Ska/Starcheck/Obsid.pm

```diff
@@ -1361,23 +1343,46 @@ sub check_star_catalog {
 	    }
 	}
 	else{
-            my $guide_edge_delta = min(($row_max - $dither_guide_y / $ang_per_pix) - $pixel_row,
-                                       $pixel_row - ($row_min + $dither_guide_y / $ang_per_pix),
-                                       ($col_max - $dither_guide_z / $ang_per_pix) - $pixel_col,
-                                       $pixel_col - ($col_min + $dither_guide_z / $ang_per_pix));
-            my $acq_edge_delta = min(($row_max - $dither_acq_y / $ang_per_pix) - $pixel_row,
-                                     $pixel_row - ($row_min + $dither_acq_y / $ang_per_pix),
-                                     ($col_max - $dither_acq_z / $ang_per_pix) - $pixel_col,
-                                     $pixel_col - ($col_min + $dither_acq_z / $ang_per_pix));
-            if (($type =~ /BOT|GUI|FID/) and ($guide_edge_delta < 0)){
-                push @warn,sprintf "$alarm [%2d] Off (padded) CCD.\n",$i;
+            my $pix_window_pad = 7; # half image size + point uncertainty + ? + 1 pixel of margin
+            my $pix_row_pad = 8;
+            my $pix_col_pad = 1;
+            my $row_lim = 512.0 - ($pix_row_pad + $pix_window_pad);
+            my $col_lim = 512.0 - ($pix_col_pad + $pix_window_pad);
+
+            my %guide_limits = ('row' => $row_lim - $dither_guide_y / $ang_per_pix,
+                                'col' => $col_lim - $dither_guide_z / $ang_per_pix);
+
+            my %pixel = ('row' => $pixel_row,
+                         'col' => $pixel_col);
+            my %pixel_sign = ('row' => ($pixel_row < 0) ? -1 : 1,
+                              'col' => ($pixel_col < 0) ? -1 : 1);
+
+            if ($type =~ /BOT|GUI|FID/){
```

Comment:
Oops.  FID on CCD checks should not use the dither-padded region.

Reply:
I think FIDs should probably have the same style of warnings and the guide stars (red, orange, yellow for off, close, and of concern distances from edge), though, so the %guide_limits should probably be renamed and moved into this block, with dither padding on BOT|GUI and no dither padding on FID.